crates/bevy_math/src/bezier.rs

```diff
@@ -0,0 +1,376 @@
+use glam::{Vec2, Vec3, Vec3A};
+
+use std::{
+    fmt::Debug,
+    iter::Sum,
+    ops::{Add, Mul, Sub},
+};
+
+/// A point in space of any dimension that supports addition and multiplication.
+pub trait Point:
+    Mul<f32, Output = Self>
+    + Add<Self, Output = Self>
+    + Sub<Self, Output = Self>
+    + Sum
+    + Default
+    + Debug
+    + Clone
+    + PartialEq
+    + Copy
+{
+}
+impl Point for Vec3 {} // 3D
+impl Point for Vec3A {} // 3D
+impl Point for Vec2 {} // 2D
+impl Point for f32 {} // 1D
```

Comment:
Not necessarily blocking, but it would be interesting to see if we should consider how this would interact with the `Animatable` trait in #4482.

Reply:
Could you explain a bit more? I don't follow.

This trait is only used to reduce code repetition for implementing Beziers in multiple dimensions/types.